src/import/codegen.ts

```diff
@@ -213,3 +213,130 @@ export function generateConstruct(typegen: TypeGenerator, def: ApiObjectDefiniti
     }
   });
 }
+
+/**
+ * Emit imports for generated helm construct
+ * @param code CodeMaker istance
+ */
+export function emitHelmHeader(code: CodeMaker) {
+  code.line('// generated by cdk8s');
+  code.line('import { Helm } from \'cdk8s\';');
+  code.line('import { Construct } from \'constructs\';');
+  code.line();
+}
+
+/**
+ * Helm Object Definition
+ */
+export interface HelmObjectDefinition {
+  /**
+   * `values.schema.json` for the helm chart
+   */
+  readonly schema: JSONSchema4 | undefined;
+  /**
+   * Chart name
+   */
+  readonly chartName: string;
+  /**
+   * Chart url
+   */
+  readonly chartUrl: string;
+  /**
+   * Chart version
+   */
+  readonly chartVersion: string;
+  /**
+   * Chart dependencies
+   */
+  readonly chartDependencies: string[];
+  /**
+   * Fully qualified name for the construct
+   */
+  readonly fqn?: string;
+}
+
+export function generateHelmConstruct(typegen: TypeGenerator, def: HelmObjectDefinition) {
+
+  const chartName = TypeGenerator.normalizeTypeName(def.chartName);
+  const schema = def.schema;
+  const repoUrl = def.chartUrl;
+  const chartVersion = def.chartVersion;
+
+  // Create custom type
+  typegen.emitCustomType(chartName, code => {
+
+    const valuesInterface = `${chartName}ValuesProps`;
+    if (schema !== undefined) {
+      // Interface for schema generated props
+      let schemaGenValuesInterface: string = 'SchemaGeneratedValues';
+      schemaGenValuesInterface = typegen.emitType(schemaGenValuesInterface, schema, def.fqn);
+
+      // Creating values interface
+      emitValuesInterface();
+
+      function emitValuesInterface() {
+        code.openBlock(`export interface ${valuesInterface} extends ${schemaGenValuesInterface}`);
```

Comment:
This complicates the code for no good reason I feel. I understand the distinction between props that come from the schema, and props that we populate - but in practice it doesn't really matter, especially for the user who might want to look at the code. Lets simplify but just putting everything in a single interface.